README.md

```diff
@@ -76,4 +76,31 @@ To ensure deterministic behavior, the pre-commit hook *does not* attempt to fix
 
 ## Ease of use
 
-For comfort, we are adding these shorthands: `npm run lint`, `npm run lf` (for `lint --fix`) and `npm run docs` (to build and serve the documentation). Apart from seeing the linter warnings when you (try to) commit or run the linter manually, your IDE should also show you these warnings in the code and fixing (the auto-fixable ones) should be offered in the context menu on hover or via Ctrl-..
+For comfort, we are adding these shorthands: `npm run lint`, `npm run lf` (for `lint --fix`) and `npm run docs` (to build and serve the documentation). Apart from seeing the linter warnings when you (try to) commit or run the linter manually, your IDE should also show you these warnings in the code and fixing (the auto-fixable ones) should be offered in the context menu on hover or via Ctrl-`.`.
+
+## The Architecture Matrix
+
+This application is built as a modularized frontend monolith (a "modulith") using vertical slices and layers as module boundaries, which are enforced using the linter.
+
+The vertical slices correspond to the different feature areas or bounded contexts within the application, such as *metadata* or *access-requests*. Additionally, we have a vertical slice called *portal* that contains all overarching features, such as the home page (which displays a metadata summary) and the user profile page (which shows the user's access requests). Another slice, called *shared*, provides UI components or utilities used across all feature areas. Authentication and user session handling are implemented in a separate slice called *auth*.
+
+The horizontal layers are named *features* (for feature components, i.e., smart components), *ui* (for presentational components, i.e., dumb components), *services* (for accessing domain objects and corresponding application logic), *models* (for interfaces of domain objects), and *utils* (for feature-specific utility functions)." The *services* layer primarily contains Angular services, while the *utils* layer includes pure pipes, guards, and custom utility functions.
+
+This results in the following architecture matrix:
+
+| portal   | metadata | access-requests | ... | auth     | shared   |
+|----------|----------|---------------- |-----|----------|----------|
+| features | features | features        | ... | features | features |
+| ui       | ui       | ui              | ... | ui       | ui       |
+| services | services | services        | ... | services | services |
+| models   | models   | models          | ... | models   | models   |
+| utils    | utils    | utils           | ... | utils    | utils    |
+
+To create a clean architecture, the following rules are checked when importing modules from the architecture matrix:
+
+- Modules within a vertical slice must only import modules from the same slice, as these slices represent bounded contexts.
+- An exception is that all vertical slices are allowed to use modules from the *shared* vertical slice.
```

Comment:
One idea that is missing is to force stuff out of shared if possible. Currently, there is no statement that we won't just move everything into shared. Do we have a definition of when that's acceptable?

Reply:
It's hard to create rules for this. It should be the task of the reviewers to check that features are put into the right vertical slice and that a new vertical slice is added when a new feature area (bounded context) is introduced.